src/framework/graphics/picker.js

```diff
@@ -76,50 +91,110 @@ class Picker {
      */
     getSelection(x, y, width = 1, height = 1) {
         const device = this.device;
+        if (device.isWebGPU) {
```

Comment:
This seems real messy. I would prefer one way of getting texture data on both (even if one implementation must emulate the other).

Reply:
This is backwards compatibility, which a lot of users depends on, and so I left it there. Deprecating this would create a breaking change for no reason for now, people can move over to the newer API without us forcing it at this stage, as that could be a larger impact change. Including the Editor.

Reply:
There is one way to get the data on both, called `Texture.read` - this is async and works on both WebGL2 and WebGPU.
Public API `Picker.getSelectionAsync` uses it, and works on both platforms.
Existing `Picker.getSelection` is left there an a synchronous WebGL2 only method, to avoid a breaking change, and prints an error on WebGPU platform, forcing people to upgrade at that point.